src/components/WarningModal/WarningModal.tsx

```diff
@@ -0,0 +1,43 @@
+import {
+  Text,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  ModalProps,
+} from "@chakra-ui/react"
+import { ModalCloseButton } from "@opengovsg/design-system-react"
+import { PropsWithChildren } from "react"
+
+interface WarningModalProps extends ModalProps {
+  displayTitle: string
+  displayText: JSX.Element
+}
+
+export const WarningModal = ({
+  isOpen,
+  onClose,
+  displayTitle,
+  displayText,
+  children,
+  ...rest
+}: PropsWithChildren<WarningModalProps>): JSX.Element => (
+  <Modal isOpen={isOpen} onClose={onClose} {...rest}>
```

Comment:
very minor nit but idt we need to explicitly specify and pass `isOpen/onClose` as those are part of `ModalProps` and the `...rest` and subsequent spread will capture them correctly

Reply:
Good point,  https://github.com/isomerpages/isomercms-frontend/pull/955/commits/021a8dd53c3dc7063bddd7d279ab4001f8d80b37